[Docs] Add documentation for Window IDs

DaVukovic opened the pull request:
As the title says...documentations for the WindowIDs.

I know, that some are missing. But I wanted to PR that so we could refere to anything from the `action()` built in here:
https://codedocs.xyz/xbmc/xbmc/page__list_of_built_in_functions.html#built_in_functions_5

The change for that section comes in another PR as well as a proper page for the ActionIDs.

I decided to use a different *dox file for that documentation, because the table would blow up the header file and I thought it's better to keep it clean.

Rechi commented:
jenkins build this please

garbear commented:
Would it be more readable if we CamelCased the window names?

DaVukovic commented:
very true, I'll change that

MartijnKaijser commented:
If we use these names internally we shouldn't :)

DaVukovic commented:
ok, then I don't change anything until there is a final decision ;)

For what I can tell, it doesn't make any difference if you use them "CamelCased" via kodi-send or all in lower case. Both is working. Just to mention that

MartijnKaijser commented:
@ronie ?

garbear commented:
window names are case-insensitive, so it's just cosmetics. IMO it comes down to whatever is more readable

DaVukovic commented:
@garbear @MartijnKaijser 

I changed the window names to be shown in CamelCase. I also added a note at the top mentioning that the CamelCase style is used only for readability and should be used lowercase internally.

I guess that´s a thing we could agree to?!

MartijnKaijser commented:
Can you squash the commits into one?

DaVukovic committed:
[Docs] Add documentation for Window IDs

Update window name after PR 14496

Merge pull request #1 from garbear/documentation

Update window name after PR 14496

DaVukovic commented:
Commits squashed and changed the wording for the note.

garbear commented on xbmc/guilib/WindowIDs.dox:
```diff
@@ -0,0 +1,145 @@
+| GameControllers         | WINDOW_DIALOG_GAME_CONTROLLERS       | 10820     | DialogGameControllers.xml           |
+| Games                   | WINDOW_GAMES                         | 10821     | MyGames.xml                         |
+| GameOSD                 | WINDOW_DIALOG_GAME_OSD               | 10822     | GameOSD.xml                         |
+| GameVideoFilter         | WINDOW_DIALOG_GAME_VIDEO_FILTER      | 10823     | DialogSettings.xml                  |
```
Should be DialogSelect.xml

garbear commented on xbmc/guilib/WindowIDs.dox:
```diff
@@ -0,0 +1,145 @@
+
+@note Window names are case-insensitive
+
+| Name                    | Definition                           | WindowID  | XML Filen                           |
```
"Filen"

garbear commented on xbmc/guilib/WindowIDs.dox:
```diff
@@ -0,0 +1,145 @@
+
+\page window_ids WindowIDs
+
+This page shows the window names, the window definition, the window ID and the specific xml file in use.
```
```
This page shows the window name, the window definition, the window ID and the specific XML file in use.
```

DaVukovic commented:
All changes done
